File: dart/test/entity/payment_sred/PaymentSredEntity_test.dart
// ignore_for_file: unused_import, unused_local_variable, non_constant_identifier_names

import 'dart:convert';
import 'dart:io';

import '../../harness.dart';
import '../../utility.dart';

import '../../../lib/BluefinTecsMerchantServicesSDK.dart';
import '../../../lib/utility/voxgig_struct.dart' as vs;

void tests() {
  describe('PaymentSredEntity', () {
    test('instance', (t) async {
      final testsdk = BluefinTecsMerchantServicesSDK.test();
      final ent = testsdk.PaymentSred();
      ok(null != ent);
    });



    test('basic', (t) async {

      final live = 'TRUE' == Platform.environment['BLUEFIN_TECS_MERCHANT_SERVICES_TEST_LIVE'];
      for (final op in ['create']) {
        if (maybeSkipControl(t, 'entityOp', 'payment_sred.' + op, live)) {
          return;
        }
      }

      final setup = basicSetup();
      // The basic flow consumes synthetic IDs and field values from the
      // fixture (entity TestData.json). Those don't exist on the live API.
      // Skip live runs unless the user provided a real ENTID env override.
      if (true == setup['syntheticOnly']) {
        t.skip('live entity test uses synthetic IDs from fixture — set BLUEFIN_TECS_MERCHANT_SERVICES_TEST_PAYMENT_SRED_ENTID JSON to run live');
        return;
      }
      final client = setup['client'];
      final struct = setup['struct'];

      final isempty = struct.isempty;
      final select = struct.select;


      // CREATE
      final payment_sred_ref01_ent = client.PaymentSred();
      dynamic payment_sred_ref01_data = setup['data']['new']['payment_sred']['payment_sred_ref01'];

      payment_sred_ref01_data = await payment_sred_ref01_ent.create(payment_sred_ref01_data);
      ok(null != payment_sred_ref01_data);


    });
  });
}


Map<String, dynamic> basicSetup([dynamic extra]) {
  final options = <String, dynamic>{};

  final entityDataFile = resolveTestPath(
      '../.sdk/test/entity/payment_sred/PaymentSredTestData.json');

  final entityDataSource = File(entityDataFile).readAsStringSync();

  final entityData = jsonDecode(entityDataSource);

  options['entity'] = entityData['existing'];

  var client = BluefinTecsMerchantServicesSDK.test(options, extra);
  final struct = client.utility().struct;
  final merge = struct.merge;
  final transform = struct.transform;

  dynamic idmap = transform(
      <dynamic>['payment_sred01', 'payment_sred02', 'payment_sred03'],
      <String, dynamic>{
        '`\$PACK`': <dynamic>[
          '',
          <String, dynamic>{
            '`\$KEY`': '`\$COPY`',
            '`\$VAL`': <dynamic>['`\$FORMAT`', 'upper', '`\$COPY`'],
          }
        ]
      });

  // Detect whether the user provided a real ENTID JSON via env var. The
  // basic flow consumes synthetic IDs from the fixture file; without an
  // override those synthetic IDs reach the live API and 4xx. Surface this
  // to the test so it can skip rather than fail.
  final idmapEnvVal =
      Platform.environment['BLUEFIN_TECS_MERCHANT_SERVICES_TEST_PAYMENT_SRED_ENTID'];
  final idmapOverridden =
      null != idmapEnvVal && idmapEnvVal.trim().startsWith('{');

  final env = envOverride({
    'BLUEFIN_TECS_MERCHANT_SERVICES_TEST_PAYMENT_SRED_ENTID': idmap,
    'BLUEFIN_TECS_MERCHANT_SERVICES_TEST_LIVE': 'FALSE',
    'BLUEFIN_TECS_MERCHANT_SERVICES_TEST_EXPLAIN': 'FALSE',
    'BLUEFIN_TECS_MERCHANT_SERVICES_APIKEY': 'NONE',
  });

  idmap = env['BLUEFIN_TECS_MERCHANT_SERVICES_TEST_PAYMENT_SRED_ENTID'];

  final live = 'TRUE' == env['BLUEFIN_TECS_MERCHANT_SERVICES_TEST_LIVE'];

  if (live) {
    client = BluefinTecsMerchantServicesSDK(merge([
      <String, dynamic>{
        'apikey': env['BLUEFIN_TECS_MERCHANT_SERVICES_APIKEY'],
      },
      extra
    ]));
  }

  final setup = <String, dynamic>{
    'idmap': idmap,
    'env': env,
    'options': options,
    'client': client,
    'struct': struct,
    'data': entityData,
    'explain': 'TRUE' == env['BLUEFIN_TECS_MERCHANT_SERVICES_TEST_EXPLAIN'],
    'live': live,
    'syntheticOnly': live && !idmapOverridden,
    'now': DateTime.now().millisecondsSinceEpoch,
  };

  return setup;
}

File: scala/sdktest/PaymentSredEntityTest.scala
// Generated basic-flow test for the payment_sred entity (model-driven;
// mirrors the java TestEntity generator). A dependency-free scala-cli test
// object driven by SdkEntityTestMain. Runs against the in-memory test
// transport seeded with the shipped PaymentSredTestData.json fixtures.

import java.util.{ArrayList, LinkedHashMap, List => JList, Map => JMap}

import voxgig.bluefintecsmerchantservicessdk.core.{Helpers, BluefinTecsMerchantServicesSDK}
import voxgig.bluefintecsmerchantservicessdk.utility.struct.Struct

object PaymentSredEntityTest {

  def run(rep: SdkTestReport): Unit = {
    rep.scope("payment_sred.instance") {
      val testsdk = BluefinTecsMerchantServicesSDK.testSDK()
      val ent = testsdk.paymentSred(null)
      rep.check("payment_sred.instance", ent != null, "expected non-null payment_sred entity")
    }

    rep.scope("payment_sred.basic") {
      val entityData = Helpers.toMapAny(SdkTestSupport.readJson(
          "../.sdk/test/entity/payment_sred/PaymentSredTestData.json"))
      val options = new LinkedHashMap[String, Object]()
      options.put("entity", entityData.get("existing"))
      val client = BluefinTecsMerchantServicesSDK.testSDK(options, null)

      val idmap = new LinkedHashMap[String, Object]()
      idmap.put("payment_sred01", "PAYMENT_SRED01")
      idmap.put("payment_sred02", "PAYMENT_SRED02")
      idmap.put("payment_sred03", "PAYMENT_SRED03")
      val now = System.currentTimeMillis()

      // CREATE
      val paymentSredRef01Ent = client.paymentSred(null)
      var paymentSredRef01Data = Helpers.toMapAny(Struct.getprop(
          Struct.getpath(entityData, "new.payment_sred"), "payment_sred_ref01"))
      val paymentSredRef01DataResult = paymentSredRef01Ent.create(paymentSredRef01Data, null)
      paymentSredRef01Data = Helpers.toMapAny(paymentSredRef01DataResult)
      rep.check("payment_sred.create.map", paymentSredRef01Data != null, "expected create result to be a map")
    }
  }
}
